src/components/Snippet.jsx

```diff
@@ -1,34 +1,48 @@
-import React, { Component } from 'react';
+import React from 'react';
+import { connect } from 'react-redux';
 import { Controlled as CodeMirror } from 'react-codemirror2';
 
 import 'codemirror/lib/codemirror.css';
 
 import Title from './common/Title';
 import Input from './common/Input';
+import Spinner from './common/Spinner';
+import * as actions from '../actions';
 
 import '../styles/Snippet.styl';
 
-class Snippet extends Component {
-  constructor() {
-    super();
+class Snippet extends React.Component {
+  constructor(props) {
+    super(props);
     this.state = { isShowEmbed: false };
     this.toggleEmbed = this.toggleEmbed.bind(this);
   }
 
+  componentDidMount() {
+    const { dispatch } = this.props;
+    const { id } = this.props.match.params;
+
+    dispatch(actions.fetchSnippet(Number(id)));
+  }
+
   toggleEmbed() {
-    this.setState({ isShowEmbed: !this.state.isShowEmbed });
+    this.setState(prevState => ({ isShowEmbed: !prevState.isShowEmbed }));
```

Comment:
Usually I'd suggest to propose such changes in separate PR because the change requires more context and rationale.